### PR TITLE
Intro hy

### PR DESCRIPTION
**Pull request by TimKam**
Looking for reviewers!

### CODE REVIEW

File: _posts/2017-11-01-hy.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  feature: /2017/squid.jpg
+  alt: Squid
+---
+
+[Hy](http://docs.hylang.org/en/stable/#) is a Lisp dialect that runs on top of - and tightly integrates with - Python.
+Considering the rise of Python to the top of many programming language rankings (for example [the one published by IEEE](https://spectrum.ieee.org/computing/software/the-2017-top-programming-languages)) and the rediscovery of functional programing patterns by the software development mainstream, I was curious if Hy succeeds in bringing the functional paradigm to one of the most popular programing ecosystems.
+
+## Getting started with Hy
+The [tutorial](http://docs.hylang.org/en/stable/tutorial.html) in the official Hy documentation excels at providing and overview of the language's most important concepts and helped me get my first code running in a matter of minutes.
+Still, the docs don't recommend and don't even mention any tooling that's available around Hy and I couldn't find such on my own.
```
Comment (rstiles) [inline _posts/2017-11-01-hy.md]: "Still, the docs don't recommend and don't even mention any tooling that's available around Hy and I couldn't find such on my own." --> "Unfortunately, the docs don't recommend or even mention any tooling that's available around Hy, and I couldn't find any on my own."

File: _posts/2017-11-01-hy.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+That's why I decided to use [PyCharm](https://www.jetbrains.com/pycharm/) as my IDE.
+This allows me to leverage Hy's Python integration to conveniently run, debug  and (potentially) test Hy code.
+Because syntactically, Hy &asymp; Lisp &asymp; Clojure, I installed a [Clojure](https://plugins.jetbrains.com/plugin/8636-clojure-kit) plugin for PyCharm and added ``*.hy`` to the list of file types the system highlights as Clojure code. I had to disable *inspections* for Clojure in the PyCharm settings to not get prompted any errors.
```
Comment (rstiles) [inline _posts/2017-11-01-hy.md]: "Because syntactically, Hy &asymp; Lisp &asymp; Clojure, I installed a [Clojure]"
This needs some clarification, I think.

"I had to disable *inspections* for Clojure in the PyCharm settings to not get prompted any errors." --> "I had to disable *inspections* for Clojure in the PyCharm settings to avoid getting error prompts."

File: _posts/2017-11-01-hy.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+We can also require a set of Hy extensions, so called [contributor modules](http://docs.hylang.org/en/stable/contrib/index.html):
+
+```clojure
+(import numpy)
+(require [hy.contrib.loop [loop]])
+```
+
+After defining the dependencies, we're ready to start with the actual neural network implementation.
+First, we define a [signoid function](https://en.wikipedia.org/wiki/Sigmoid_function) as our activation function.
+Like in Lisp, in Hy operators are proper variadic functions and of course we need to use tons of parentheses:
```
Comment (rstiles) [inline _posts/2017-11-01-hy.md]: "Like in Lisp, in Hy operators are proper variadic functions and of course we need to use tons of parentheses" --> "Just like Lisp, operators in Hy are proper variadic functions--and, of course, we need to use tons of parentheses"

File: _posts/2017-11-01-hy.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+You can also find the code on [GitHub](https://github.com/TimKam/hylang_gradient_descent).
+
+For advanced, production-like use cases, it is possible to utilize further tools and services that support or were specifically created for Python.
+For example, to enable continuous integration we could run unit tests with  [pytest](https://docs.pytest.org/en/latest/) and set up a continuous integration workflow with [Travis](https://travis-ci.org/) or [Circle CI](https://circleci.com/).
+
+## Assessment: what's Hy good for?
+Hy is a creative attempt to introduce a highly praised, but sparsely adopted functional programing language to the Python community.
+Hy's Lisp-y syntax and its tight Python integration compensates to some degree for lack of tooling.
+This means that in most use cases, you can't use Hy on it's own, but need to use Python to manage aspects like testing and dependency management and tooling for Lisp or other Lisp-based languages for syntax highlighting.
```
Comment (rstiles) [inline _posts/2017-11-01-hy.md]: "This means that in most use cases, you can't use Hy on it's own, but need to use Python to manage aspects like testing and dependency management and tooling for Lisp or other Lisp-based languages for syntax highlighting."

Try something like this instead:
"This means that, in most use cases, you can't use Hy on its own. Instead, you need to use Python to manage aspects like testing and dependency management, along with tooling for Lisp or other Lisp-based languages for syntax highlighting"

File: _posts/2017-11-01-hy.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+However, static code analysis isn't properly possible and will eventually require Hy-specific tooling.
```
Comment (rstiles) [inline _posts/2017-11-01-hy.md]: "However, static code analysis isn't properly possible and will eventually require Hy-specific tooling."

"Properly possible" sounds a little odd in English. Maybe "Proper static code analysis isn't possible and will eventually require..." or similar phrasing.

File: _posts/2017-11-01-hy.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+We can also require a set of Hy extensions, so called [contributor modules](http://docs.hylang.org/en/stable/contrib/index.html):
+
+```clojure
+(import numpy)
+(require [hy.contrib.loop [loop]])
+```
+
+After defining the dependencies, we're ready to start with the actual neural network implementation.
+First, we define a [signoid function](https://en.wikipedia.org/wiki/Sigmoid_function) as our activation function.
+Like in Lisp, in Hy operators are proper variadic functions and of course we need to use tons of parentheses:
```
Comment (hilton) [inline _posts/2017-11-01-hy.md]: 😂

File: _posts/2017-11-01-hy.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+                 1 (- (* 2 (.random.random numpy [4 1])) 1)})
+```
+
+Finally, we run the training function with the input parameters we specified above:
+
+```clojure
+(print (+ "Output after training \n\r" (str (train synapses input-data 60000))))
+```
+
+You can also find the code on [GitHub](https://github.com/TimKam/hylang_gradient_descent).
```
Comment (hilton) [inline _posts/2017-11-01-hy.md]: This seems buried here. How above moving this paragraph to immediately after the introduction, before the _ Getting started with Hy_ heading. That way people who want to read code first, can.

File: _posts/2017-11-01-hy.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+Hy is syntactically more attractive for buffs who already know expert languages like Clojure.
+The many parentheses might put off folks who are new to functional programming or know it primarily from modern JavaScript.
+Some compromises were made to also support Pythonic syntax, most  notably *dot notation*:
+
+```clojure
+  (numpy.mean [1 5]) ;; returns 3
+```
+
+Still, I personally would like Hy to draw more inspiration from Python’s syntax and support (significant) indentation instead of parentheses.
```
Comment (hilton) [inline _posts/2017-11-01-hy.md]: This is probably the best idea on the whole of _tech.signavio.com_!

File: _posts/2017-11-01-hy.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+
+[PyCharm](https://www.jetbrains.com/pycharm/) allows me to leverage Hy's Python integration to conveniently run, debug  and (potentially) test Hy code.
+Because syntactically, Hy &asymp; Lisp &asymp; Clojure, I installed a [Clojure](https://plugins.jetbrains.com/plugin/8636-clojure-kit) plugin for PyCharm and added ``*.hy`` to the list of file types the system highlights as Clojure code. I had to disable *inspections* for Clojure in the PyCharm settings to not get prompted for any errors.
+When I had trouble with Hy's syntax while programming, I used ``Lisp`` or ``Clojure`` instead of ``Hy`` in my search engine queries.
+
+## Implementing a basic neural network with Hy
+Now, as we have the basics covered, let's write some code.
+As an example, I picked a basic neural network (gradient descent) as presented by Siraj Raval in [this YouTube video](https://www.youtube.com/watch?v=h3l4qz76JhQ) and implemented it in Hy instead of Python.
+
+In Hy, we can import any Python library - in our example *numpy*.
```
Comment (hilton) [inline _posts/2017-11-01-hy.md]: Maybe mention what _numpy_ is for (numerical programming?) for those of us who don't use Python.

File: _posts/2017-11-01-hy.md [hunk 10]
```diff
@@ -9,35 +9,70 @@ image:
...
         (return (* x (- 1 x)))
         (return (/ 1 (+ 1 (.exp numpy (- x)))))))
+```
+
+And here's the code that does the actual training.
+To implement the training loop, we need the extension we required above.
+
+```clojure
+;; training
+(defn train [synapses input-data index]
```
[the remark this answers is not in the document]
Comment (TimKam) [reply _posts/2017-11-01-hy.md]: I use the dictionary because it's mutable.

File: _posts/2017-11-01-hy.md [hunk 11]
```diff
@@ -9,35 +9,70 @@ image:
...
 (import numpy)
 (require [hy.contrib.loop [loop]])
+```
+
+After defining the dependencies, we're ready to start with the actual neural network implementation.
+First, we define a [signoid function](https://en.wikipedia.org/wiki/Sigmoid_function) as our activation function.
+Like in Lisp, in Hy operators are proper variadic functions and of course we need to use tons of parentheses:
 
+```clojure
 (defn nonlin [x &optional (derive False)] (
```
[the remark this answers is not in the document]
Comment (TimKam) [reply _posts/2017-11-01-hy.md]: I agree, but want to keep it as-is to have the optional variable and implicit `else` in the code :-)
